Саитов Владислав

nmkname opened the pull request.

nmkname committed:
Первый вариант решения

honest-hrundel commented:
:tomato: Пройдено тестов 3 из 16

nmkname committed:
Второй вариант решения

honest-hrundel commented:
:tomato: Пройдено тестов 3 из 16

nmkname committed:
Третий

honest-hrundel commented:
:tomato: Не пройден линтинг или базовые тесты

nmkname committed:
Четвертый?

honest-hrundel commented:
:tomato: Не пройден линтинг или базовые тесты

nmkname committed:
Четвертый.1

honest-hrundel commented:
:tomato: Не пройден линтинг или базовые тесты

nmkname committed:
Четвертый.2

honest-hrundel commented:
:tomato: Пройдено тестов 3 из 16

nmkname committed:
Пятый

honest-hrundel commented:
:tomato: Пройдено тестов 3 из 16

nmkname committed:
Пятый.2

nmkname committed:
Update robbery.js

honest-hrundel commented:
:tomato: Не пройден линтинг или базовые тесты

honest-hrundel commented:
:tomato: Пройдено тестов 6 из 16

nmkname committed:
Пятый.3

honest-hrundel commented:
:tomato: Пройдено тестов 6 из 16

nmkname committed:
n-th solution way

honest-hrundel commented:
:tomato: Пройдено тестов 6 из 16

nmkname committed:
Может здесь ошибка

honest-hrundel commented:
:tomato: Пройдено тестов 6 из 16

nmkname committed:
LET ME IN

honest-hrundel commented:
:tomato: Пройдено тестов 6 из 16

nmkname committed:
sssssss

honest-hrundel commented:
:tomato: Пройдено тестов 6 из 16

nmkname committed:
Completely rebuilded code, otherwise - give up

honest-hrundel commented:
:tomato: Пройдено тестов 15 из 16

nmkname committed:
Even extra one is done

honest-hrundel commented:
:tomato: Пройдено тестов 17 из 19

nmkname committed:
no extra one

honest-hrundel commented:
:green_apple: Пройдено тестов 16 из 16

nmkname committed:
gonna solve it

honest-hrundel commented:
:tomato: Пройдено тестов 17 из 19

nmkname committed:
Первый вариант

honest-hrundel commented:
:tomato: Не пройден линтинг или базовые тесты

nmkname committed:
Первый ???

honest-hrundel commented:
:tomato: Пройдено тестов 18 из 19

nmkname committed:
Doubtly its gonna change sth

honest-hrundel commented:
:tomato: Пройдено тестов 18 из 19

nmkname committed:
THIS GOTTA BE CORRECT

honest-hrundel commented:
:tomato: Пройдено тестов 18 из 19

nmkname committed:
Gave up

honest-hrundel commented:
:green_apple: Пройдено тестов 16 из 16

chipolinka commented:
ты всё-таки без \* сдаёшь?)

chipolinka commented on robbery.js:
```diff
@@ -16,35 +17,158 @@ exports.isStar = true;
+        }
+    }
+
+    return false;
```
Странно, что функция возвращает либо список, либо `false`. Логичнее было бы при отсутствии пересечений возвращать `null`.

chipolinka commented on robbery.js:
```diff
@@ -16,35 +17,158 @@ exports.isStar = true;
+    for (var i = index || 1; i < busyList.length; i++) {
+        if (busyList[i][0] - busyList[i - 1][1] >= duration) {
+
+            return [busyList[i - 1][1], i - 1];
```
Почему именно список возвращается?

nmkname replied:
Это чтобы дополнительное задание можно было сделать. Костыль такой небольшой соорудил в общем(возвращается список из нужного времени и индекса этого времени в списке занятости, чтобы потом его использовать для нахождения более позднего времени).

chipolinka replied:
Тогда уже можно избавиться от этого костыля

chipolinka commented on robbery.js:
```diff
@@ -16,35 +17,158 @@ exports.isStar = true;
+    return result;
+}
+
+function getIntersection(busyList, duration, index) {
```
Непонятно, что такое `busyList`. Если список занятости, то почему это список списков?

chipolinka replied:
И индекс чего передается?

nmkname replied:
Алгоритм такой: мы объединяем все промежутки, когда не работает банк и когда заняты друзья в набор интервалов, который мы потом сортируем, объединяем в интервалы, которые между собой никак не пересекаются и расстояние между ними и есть время, когда совпадает время работы банка и свобода действий у друзей. А про индекс я написал выше. Это я сейчас уберу)

chipolinka replied:
Намекаю, что нужны более понятные названия)

chipolinka commented on robbery.js:
```diff
@@ -16,35 +17,158 @@ exports.isStar = true;
+        Number(workingHours.to.slice(3, 5));
+    result.push([0, startWorking], [endWorking, startWorking + 60 * 24],
+        [endWorking + 60 * 24, startWorking + 2 * 60 * 24],
+        [endWorking + 2 * 60 * 24, 3 * 60 * 24 - 1]);
```
Ух, как много констант, которые можно вынести!

chipolinka replied:
И, честно говоря, ничего не понятно, что тут происходит. Перепиши понятнее: разнеси по переменным/функциям или ещё как-нибудь.

nmkname replied:
Я же правильно понял, что все операции с числами надо заменить на константы? А вот по функциям я не знаю как это распределить. Просто это все цельный пуш и аргументы передавать некуда. Это надо в одном действии делать. Я могу вынести саму операцию пуша в отдельную функцию, если это надо

chipolinka commented on robbery.js:
```diff
@@ -16,35 +17,158 @@ exports.isStar = true;
+}
+
+function getEarliest(busyList, indexOfRob, duration) {
+    var newTime;
```
Почему ты объявляешь переменную так?) Можно же сразу `var newTime = ...`

nmkname replied:
Это да. Помню как это еслинт ругался на то, что переменные в варе объявляю(хотя здесь я их объявлял и в варе тоже) и вот такая привычка иногда всплывает)

chipolinka commented on robbery.js:
```diff
@@ -16,35 +17,158 @@ exports.isStar = true;
+    newTime = busyList[indexOfRob][1] + 30;
+    for (var i = indexOfRob; i < busyList.length; i++) {
+        if (busyList[i - 1][1] <= newTime && busyList[i][0] >= newTime) {
+            var newList;
```
Название не говорящее.

nmkname replied:
Это я убрал уже)

chipolinka commented on robbery.js:
```diff
@@ -16,35 +17,158 @@ exports.isStar = true;
+            }
+            robberyTime = newRobberyTime;
+
+            return true;
```
Много лишних пустых строк. И попробуй это переписать намного короче. Можно, например, тернарные операторы использовать)

nmkname replied:
И это тоже

nmkname replied:
Я же уже не могу звездочку досдать, чтобы был полный балл за задание?

chipolinka commented on robbery.js:
```diff
@@ -16,35 +17,158 @@ exports.isStar = true;
         exists: function () {
-            return false;
+
+            return !(robberyTime === false);
```
Ой, а можно как-то проще, а не сравнивать с `false`?

nmkname replied:
У меня сначала был вариант if (robberyTime) {return true}; return false, но в одну строчку мне казалось это выглядит поприятнее

chipolinka commented on robbery.js:
```diff
@@ -4,7 +4,8 @@
  */
-exports.isStar = true;
+exports.isStar = false;
+var WEEKDAY_DICT = ['ПН', 'ВТ', 'СР'];
```
Но это же не словарь...

chipolinka commented:
Привет! Я пока посмотрела только стиль кода. Не стала писать повторяющиеся комментарии, так что проверь на каждое замечание, есть ли ещё где-то косяки в коде, кроме прокомментированного :)
И если ты всё-таки решил не делать задание со звёздочкой, то может, уберёшь его из кода?
Жду исправлений :)

chipolinka commented:
🍅

nmkname committed:
Here we go

Здесь я не убрал пока что тот пуш большой со временем банка, но если надо - сделаю

honest-hrundel commented:
:green_apple: Пройдено тестов 16 из 16

nmkname committed:
Убрал пробелы еще

honest-hrundel commented:
:tomato: Не пройден линтинг или базовые тесты

nmkname committed:
Ничего не изменил?

honest-hrundel commented:
:tomato: Не пройден линтинг или базовые тесты

nmkname committed:
Вот теперь то должно

honest-hrundel commented:
:tomato: Не пройден линтинг или базовые тесты

nmkname committed:
quite hilarious

honest-hrundel commented:
:green_apple: Пройдено тестов 16 из 16

chipolinka commented on robbery.js:
```diff
@@ -4,7 +4,11 @@
-exports.isStar = true;
+exports.isStar = false;
+var CORRECT_WEEKDAY = ['ПН', 'ВТ', 'СР'];
+var TUESDAY_MIN = 1440;
```
Что это такое? Оо

nmkname replied:
Это время, чтобы перейти на следующий день. А как его еще назвать?

chipolinka commented on robbery.js:
```diff
@@ -4,7 +4,11 @@
+exports.isStar = false;
+var CORRECT_WEEKDAY = ['ПН', 'ВТ', 'СР'];
+var TUESDAY_MIN = 1440;
+var WEDNESDAY_MIN = 2880;
```
А это? Почему "этого" в 2 раза больше? оО

chipolinka commented on robbery.js:
```diff
@@ -4,7 +4,11 @@
+var CORRECT_WEEKDAY = ['ПН', 'ВТ', 'СР'];
+var TUESDAY_MIN = 1440;
+var WEDNESDAY_MIN = 2880;
+var WEDNESDAY_END_MIN = 4320 - 1;
```
А это? И где тогда уж `MONDAY_MIN`? 

![cvwucihweaah-cj](https://cloud.githubusercontent.com/assets/11631479/19795064/020b2742-9cfa-11e6-8f40-482e39f25d20.jpeg)

nmkname replied:
Я могу добавить константу 0, которая и будет понедельником, так как с него и начинается

chipolinka commented on robbery.js:
```diff
@@ -16,35 +20,134 @@ exports.isStar = true;
 exports.getAppropriateMoment = function (schedule, duration, workingHours) {
     console.info(schedule, duration, workingHours);
+    var busyList = [];
+    var gmt = Number(workingHours.from.slice(6));
```
Постарайся сделать это [без объектной обёртки](http://codeharmony.ru/materials/132)

chipolinka commented on robbery.js:
```diff
@@ -16,35 +20,134 @@ exports.isStar = true;
          */
+
         exists: function () {
+            if (robberyTime) {
```
Слабо переписать это в одну строчку? :wink:

chipolinka commented on robbery.js:
```diff
@@ -16,35 +20,134 @@ exports.isStar = true;
         format: function (template) {
-            return template;
+            if (robberyTime) {
+                var timeDay = Math.floor(robberyTime[0] / 60);
```
Константы выноси) И так по всему коду.

chipolinka commented on robbery.js:
```diff
@@ -16,35 +20,134 @@ exports.isStar = true;
          * Попробовать найти часы для ограбления позже [*]
          * @star
-         * @returns {Boolean}
+         * @returns {Boolean}`
```
Кавычечка

chipolinka commented on robbery.js:
```diff
@@ -16,35 +20,134 @@ exports.isStar = true;
          */
+
         tryLater: function () {
+
```
Отступ

chipolinka commented on robbery.js:
```diff
@@ -16,35 +20,134 @@ exports.isStar = true;
+
+function compareNumb(a, b) {
+
+    return a[0] - b[0];
```
Если это сравнение чисел, то почему ты берешь у них первый элемент? Это же не списки? :)
И да, отступ.

nmkname replied:
Немного не понял про отступ

chipolinka commented on robbery.js:
```diff
@@ -16,35 +20,134 @@ exports.isStar = true;
+function unionOfTime(timeArray) {
+    timeArray.sort(compareNumb);
+    var result = [];
+    var leftp;
```
`var leftp = ...`
И непонятное название

chipolinka commented on robbery.js:
```diff
@@ -16,35 +20,134 @@ exports.isStar = true;
+    var result = [];
+    var leftp;
+    var rightp;
+    leftp = timeArray[0][0];
```
Не проще ли в `timeArray` хранить не список списков, а список объектов? Допустим, 

``` javascript
{
    start = ...,
    end = ...
}
```

Так будет читабельнее и понятнее.

chipolinka commented on robbery.js:
```diff
@@ -4,7 +4,11 @@
  * Сделано задание на звездочку
```
Готова поспорить

chipolinka commented on robbery.js:
```diff
@@ -16,35 +20,134 @@ exports.isStar = true;
+    return result;
+}
+
+function getString(stringTime, gmt) {
```
В этой функции куча констант, захардкоженых дней (а вдруг они сдвинут всё на день?), объектных обёрток и магии. Нехорошо как-то выходит. Надо переделать.

chipolinka replied:
Название функции надо поконкретнее.

chipolinka commented on robbery.js:
```diff
@@ -16,35 +20,134 @@ exports.isStar = true;
+        (gmt - Number(stringTime.slice(9)))) * 60 + Number(stringTime.slice(6, 8)) + currentDay;
+}
+
+function getWorkTime(workingHours) {
```
В этой функции замечено нарушение DRY. И куча констант. И куча непонятных констант.

chipolinka commented:
Не везде исправил замечания с первой проверки. И добавились новые.

chipolinka commented:
:tomato:

nmkname committed:
Another one

honest-hrundel commented:
:tomato: Пройдено тестов 0 из 16

nmkname commented:
ахахахахахахахаххахахаах

nmkname committed:
Просто проверка работоспособности объектов

honest-hrundel commented:
:tomato: Пройдено тестов 15 из 16

nmkname committed:
Может так?

honest-hrundel commented:
:tomato: Пройдено тестов 15 из 16

nmkname committed:
Тааааак

honest-hrundel commented:
:tomato: Не пройден линтинг или базовые тесты

nmkname committed:
Попытка найти ошибку

honest-hrundel commented:
:green_apple: Пройдено тестов 16 из 16

nmkname committed:
Немного изменил кое-что

honest-hrundel commented:
:tomato: Пройдено тестов 15 из 16

nmkname committed:
А теперь где ошибка?

honest-hrundel commented:
:tomato: Пройдено тестов 15 из 16

nmkname committed:
???????????///

honest-hrundel commented:
:tomato: Пройдено тестов 15 из 16

nmkname committed:
Может здесь?

honest-hrundel commented:
:tomato: Пройдено тестов 15 из 16

nmkname committed:
Ничего не понимаю уже

honest-hrundel commented:
:tomato: Не пройден линтинг или базовые тесты

nmkname committed:
А так?

honest-hrundel commented:
:green_apple: Пройдено тестов 16 из 16

nmkname committed:
В этом что ли?

honest-hrundel commented:
:tomato: Пройдено тестов 15 из 16

nmkname committed:
Проблема с нулём что ли?

honest-hrundel commented:
:tomato: Пройдено тестов 15 из 16

nmkname committed:
Внес несколько изменений

honest-hrundel commented:
:tomato: Пройдено тестов 15 из 16

nmkname committed:
Исправил все, кроме getString

honest-hrundel commented:
:green_apple: Пройдено тестов 16 из 16

chipolinka commented on robbery.js:
```diff
@@ -1,50 +1,141 @@
+}
+
+function getString(stringTime, gmt) {
+    return (parseInt(stringTime.slice(3, 5)) +
```
Кучу парсИнтов и слайсов лучше вынести в переменные с говорящими названиями. А то не понятно, что тут происходит.

chipolinka replied:
И в парсИнт нужно указывать систему счисления во избежание ошибок.

chipolinka commented on robbery.js:
```diff
@@ -1,50 +1,141 @@
+        parseInt(workingHours.slice(3, 5));
+}
+
+function getWorkTime(workingHours) {
```
WorkTime чего?) И что конкретно лежит в result? Если интервалы, то так и назови.

chipolinka commented on robbery.js:
```diff
@@ -1,50 +1,141 @@
+        DAYS.indexOf(stringTime.slice(0, 2)) * HOURS * MINUTES;
+}
+
+function parseWorkTime(workingHours) {
```
В эту функцию же не только часы передаются (workingHours)? Да и парсится не только рабочее время, а всё, какое захочешь :)

nmkname replied:
Как раз-таки только они и передаются, чтобы не нарушать правило dry

chipolinka commented on robbery.js:
```diff
@@ -1,50 +1,141 @@
+var DAYS = ['ПН', 'ВТ', 'СР'];
+var HOURS = 24;
+var MINUTES = 60;
+var TWO_DIGITS = 10;
```
10, мне кажется можно не выносить в константу. А сделать функцию, которая бы принимала число на вход и возвращала его строковое значение (с прибавленным впереди нулём, если надо)

chipolinka commented on robbery.js:
```diff
@@ -1,50 +1,141 @@
+    var busyTimeList = [];
+    var gmt = parseInt(workingHours.from.slice(6));
+    var robberyTime;
+    busyTimeList = busyTimeList.concat(getCorrectTime(schedule.Danny, gmt),
```
А если к банде добавится, не дай бог, еще парочка бандитов -- у тебя они не учтутся :)

chipolinka commented on robbery.js:
```diff
@@ -1,50 +1,141 @@
+    return result;
+}
+
+function unionOfTime(timeArray) {
```
Тут же речь не совсем о времени, а об интервалах, верно? Названия должны соответствовать.

chipolinka commented:
:tomato:

nmkname committed:
Исправил все, кроме parseWork(парсит только банк)

honest-hrundel commented:
:green_apple: Пройдено тестов 16 из 16

chipolinka commented on robbery.js:
```diff
@@ -1,50 +1,150 @@
     console.info(schedule, duration, workingHours);
+    var busyTimeList = [];
+    var gmt = parseInt(workingHours.from.slice(6), 10);
+    var robberyTime;
```
Лучше вообще везде избавиться от такой инициализации:
1. Она далеко от первого использования переменной
2. Непонятно, что это за тип
3. Почему нельзя просто написать `var robberyTime = getIntersection(busyTimeList, duration);`? :)

nmkname replied:
Робери тайм и есть время, то есть число. Я изменю название переменной на robberyMinutesTime. Все вары лучше же писать вначале, поэтому я решил сделать так. Но это я тоже исправлю.

chipolinka commented on robbery.js:
```diff
@@ -1,50 +1,150 @@
     };
 };
+
+function lessThanTen(Numb) {
```
Название переменной с маленькой буквы

chipolinka commented on robbery.js:
```diff
@@ -1,50 +1,150 @@
+    var workIntervals = [];
+    var startWorking = parseBankWorkTime(workingHours.from);
+    var endWorking = parseBankWorkTime(workingHours.to);
+    workIntervals.push(
```
Здесь напрашивается заполнение через цикл

nmkname replied:
Нам все равно придется делать на два шага меньше, так как первый и последний нельзя задать в цикле без условий, а тут мы знаем точное количество дней, так что строчек кода выйдет столько же, а для трех дней даже меньше.

chipolinka commented on robbery.js:
```diff
@@ -1,50 +1,150 @@
+function getIntersection(busyTimeList, duration) {
+    for (var i = 1; i < busyTimeList.length; i++) {
+        if (busyTimeList[i].start - busyTimeList[i - 1].end >= duration) {
+
```
Линтер пропустит, если здесь убрать пустую строку?

nmkname replied:
Пропустит

chipolinka commented:
:tomato:

nmkname committed:
Исправил все, кроме заполнения время работы

honest-hrundel commented:
:green_apple: Пройдено тестов 16 из 16

chipolinka commented:
🚀

gogoleff commented:
🚀

gogoleff commented:
🚀

nmkname commented:
Здравствуйте. Я только уже пол балла смогу получить за задание? Просто я так понял, что произошло непредвиденное и сейчас таким же, как я назначают новых менторов, а дедлайн уже прошел.

gogoleff commented:
Мы немного перераспредели нагрузку по менторам. 
Задачу на ментора перевели только 1 день назад, так что у него ещё 1 день есть.

gogoleff commented:
Лучше задачу впритык к дедлайнам не сдавать, а стараться хотя бы за дня 4

nmkname commented:
Это да, я сам виноват конечно, что так получилось. Досдал я ее только во вторник, в 3 часа. Просто так бы у меня может было два дня на рефакторинг, а в итоге получится один.

msmirnov commented on robbery.js:
```diff
@@ -1,50 +1,148 @@
 /**
  * @param {Object} schedule – Расписание Банды
- * @param {Number} duration - Время на ограбление в минутах
+ * @param {parseInt} duration - Время на ограбление в минутах
```
`parseInt`?

nmkname replied:
это когда я заменял объектную обертку намбер на парсИнт везде, не заметил, что это изменилось

msmirnov commented on robbery.js:
```diff
@@ -1,50 +1,148 @@
 exports.getAppropriateMoment = function (schedule, duration, workingHours) {
     console.info(schedule, duration, workingHours);
+    var busyTimeList = [];
+    var gmt = parseInt(workingHours.from.slice(6), 10);
```
Много где используется метод `slice`, глядя на такой код кажется непонятным, что делает та или иная строчка. Предлагаю реализовать метод, который будет парсить строчки типа "10:00+5" и возвращать нужные значения (время, зону)

nmkname replied:
У меня такая сначала была проблема, когда я все в одной функции делал, а теперь у меня там как раз-таки вынесена в отдельные функции работа со `slice`. getString работает со временем друзей, а parseBankWorkTime - с банком. А вначале получение часового пояса банка наверное не стоит класть в цикл постоянный, раз оно постоянное.

msmirnov commented on robbery.js:
```diff
@@ -1,50 +1,148 @@
+    }
+    busyTimeList = busyTimeList.concat(getBankWorkTime(workingHours));
+    busyTimeList.sort(compareStart);
+    busyTimeList = unionOfIntervals(busyTimeList);
```
Результат функции `unionOfIntervals` я бы записал в другую переменную с другим названием. Получается, что ты постоянно изменяешь `busyTimeList`, при этом название остается старым. Получается путаница. А еще в этом месте я бы посмотрел в пользу использования `reduce` или `forEach`.

nmkname replied:
Я же правильно понял, что reduce нужен в нижеупомянутых функциях?

msmirnov commented on robbery.js:
```diff
@@ -1,50 +1,148 @@
+
         exists: function () {
-            return false;
+            return robberyMinutesTime !== null;
```
Зачем здесь именно проверка на `null`, почему не обойтись `return ! robberyMinutesTime;`?

nmkname replied:
Я тоже думал над этим, но тогда придется делать отдельную проверку на ноль, а в тестах это есть - долго ошибку не мог найти.

msmirnov replied:
Ок

msmirnov commented on robbery.js:
```diff
@@ -1,50 +1,148 @@
+                var timeMin = robberyMinutesTime % MINUTES;
+
+                return template
+                        .replace(/%DD/, DAYS[(Math.floor(timeDay / HOURS))])
```
Зачем `Math.floor(timeDay / HOURS)` обёрнуто в круглые скобки? И я бы вынес это в переменную, так не читаемо.

msmirnov commented on robbery.js:
```diff
@@ -1,50 +1,148 @@
     };
 };
+
+function lessThanTen(numb) {
```
Название функции не говорит о том, что она делает. Попробуй сделать название глаголом.

msmirnov commented on robbery.js:
```diff
@@ -1,50 +1,148 @@
+    return (numb < 10) ? ('0' + numb) : numb;
+}
+
+function compareStart(a, b) {
```
Функция используется один раз, может её заинлайнить в `.sort`?

msmirnov commented on robbery.js:
```diff
@@ -1,50 +1,148 @@
+            end: getString(friendSchedule[i].to, gmt) });
+    }
+
+    return intervals;
```
Цепочку `var - for - return` можно переписать с использованием `reduce`.

msmirnov commented on robbery.js:
```diff
@@ -1,50 +1,148 @@
+function getCorrectTime(friendSchedule, gmt) {
+    var intervals = [];
+    for (var i = 0; i < friendSchedule.length; i++) {
+        intervals.push({ start: getString(friendSchedule[i].from, gmt),
```
Форматирование - не очень, принято

```
intervals.push({
   start: getString(friendSchedule[i].from, gmt),
   end: getString(friendSchedule[i].to, gmt)
});
```

msmirnov commented on robbery.js:
```diff
@@ -1,50 +1,148 @@
+    var thiefGmt = gmtBank - parseInt(stringTime.slice(9), 10);
+    var thiefBusyMinutes = parseInt(stringTime.slice(6, 8), 10);
+    var currentDay = DAYS.indexOf(stringTime.slice(0, 2)) * HOURS;
+
```
Здесь то же самое замечание про `slice`.

msmirnov commented on robbery.js:
```diff
@@ -1,50 +1,148 @@
+        { start: endWorking + 2 * MINUTES * HOURS, end: 3 * MINUTES * HOURS - 1 }
+        );
+
+    return workIntervals;
```
Зачем `workIntervals` объявляется, заполняется и возвращается в трех разных местах? Можно сделать просто `return [{},{},{},{}];`

msmirnov commented on robbery.js:
```diff
@@ -1,50 +1,148 @@
+    }
+    intervals.push({ start: previousStart, end: previousEnd });
+
+    return intervals;
```
Операции с intervals - в `reduce`

msmirnov commented:
🍅

nmkname committed:
Исправил все, кроме slice и exist(ответил)

honest-hrundel commented:
:green_apple: Пройдено тестов 16 из 16

msmirnov commented on robbery.js:
```diff
@@ -16,35 +15,133 @@ exports.isStar = true;
+    var keys = Object.keys(schedule);
+    for (var i = 0; i < keys.length; i++) {
+        busyTimeList = busyTimeList.concat(getCorrectTime(schedule[keys[i]], gmt));
+    }
```
Имел в виду, что тут можно значительно сократить код, если использовать конструкцию типа

```
var busyTimeList = Object.keys(schedule).reduce(function () {}, []);
```

msmirnov commented on robbery.js:
```diff
@@ -16,35 +15,133 @@ exports.isStar = true;
+    var startWorking = parseBankWorkTime(workingHours.from);
+    var endWorking = parseBankWorkTime(workingHours.to);
+
+    return [{ start: 0, end: startWorking },
```
С форматированием беда.

```
[
   {},
   {},
   {},
   {}
]
```

msmirnov commented:
🍅

nmkname committed:
Убрал getWorkTime - заменил на редьюс с мапом

honest-hrundel commented:
:green_apple: Пройдено тестов 16 из 16
